pallets/schemas/src/lib.rs

```diff
@@ -174,6 +181,19 @@ pub mod pallet {
 		OptionQuery,
 	>;
 
+	/// Storage for SchemaV2
+	/// - Key: Schema Id
+	/// - Value: [`SchemaV2`](SchemaV2)
+	#[pallet::storage]
+	#[pallet::getter(fn get_schema_v2)]
+	pub(super) type SchemasV2<T: Config> = StorageMap<
```

Comment:
Just a thought: If we rename the current/original `Schemas` storage to `SchemasV1` and call the new one `Schemas`, it will ensure that any existing references will use the new storage (and eliminate the need for post-migration renaming/code modification). (Will still need to delete the V1 storage post-migration, but that's all).